src/Analyzers/MSTest.Analyzers.CodeFixes/PreferTestInitializeOverConstructorFixer.cs

```diff
@@ -0,0 +1,115 @@
+[Shared]
+public sealed class PreferTestInitializeOverConstructorFixer : CodeFixProvider
+{
+    public override ImmutableArray<string> FixableDiagnosticIds { get; }
+        = ImmutableArray.Create(DiagnosticIds.PreferTestInitializeOverConstructorRuleId);
+
+    public override FixAllProvider GetFixAllProvider()
+        // See https://github.com/dotnet/roslyn/blob/main/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+        => WellKnownFixAllProviders.BatchFixer;
+
+    public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        SyntaxNode root = await context.Document.GetRequiredSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+        Diagnostic diagnostic = context.Diagnostics[0];
+        TextSpan diagnosticSpan = diagnostic.Location.SourceSpan;
+
+        SyntaxToken syntaxToken = root.FindToken(diagnosticSpan.Start);
+        if (syntaxToken.Parent is null)
+        {
+            return;
+        }
+
+        // Find the constructor declaration identified by the diagnostic.
+        ConstructorDeclarationSyntax constructorDeclaration = syntaxToken.Parent.AncestorsAndSelf().OfType<ConstructorDeclarationSyntax>().FirstOrDefault();
+        if (constructorDeclaration == null)
+        {
+            return;
+        }
+
+        context.RegisterCodeFix(
+            CodeAction.Create(
+                CodeFixResources.ReplaceWithTestInitializeFix,
+                c => ReplaceConstructorWithTestInitializeAsync(context.Document, constructorDeclaration, c),
+                nameof(PreferTestInitializeOverConstructorFixer)),
+            diagnostic);
+    }
+
+    private static async Task<Document> ReplaceConstructorWithTestInitializeAsync(Document document, ConstructorDeclarationSyntax constructorDeclaration, CancellationToken cancellationToken)
+    {
+        DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+
+        // Find the class containing the constructor
+        if (constructorDeclaration.Parent is ClassDeclarationSyntax containingClass)
+        {
+            // Check if a TestInitialize method already exists
+            MethodDeclarationSyntax? existingTestInitialize = containingClass.Members
+                .OfType<MethodDeclarationSyntax>()
+                .FirstOrDefault(m => m.AttributeLists
+                    .SelectMany(attrList => attrList.Attributes)
+                    .Any(attr => attr.Name.ToString().Contains("TestInitialize")));
+
+            // Move the body of the constructor
+            BlockSyntax? constructorBody = constructorDeclaration.Body;
+
+            if (existingTestInitialize != null)
+            {
+                // If TestInitialize exists, append the constructor body to it
+                StatementSyntax[]? constructorStatements = constructorBody?.Statements.ToArray();
```

Comment:
What's the type of `Statements`? Do you really need to `ToArray`?

Reply:
it's a `SyntaxList` so it needs the `ToArray()`